src/controller/content/content.controller.ts

```diff
@@ -599,4 +602,79 @@ export class ContentController {
     }
     return this.contentService.getAssessmentDetailsOfOpenEnded(assessmentOutsourseId, userId);
   }
+
+  
+  @Post('/createQuestionType')
+  @ApiOperation({ summary: 'Create a Question Type for the form' })
+  @ApiBearerAuth()
+  async createQuestionType(@Body() questionType: CreateTypeDto) {
+    const res = await this.contentService.createQuestionType(questionType);
+    return res;
+  }
+
+  @Get('/allQuestionType')
+  @ApiOperation({ summary: 'Get all the available Question Types' })
+  @ApiBearerAuth()
+  async getAllQuestionTypes() {
+    const res = await this.contentService.getAllQuestionTypes();
+    return res;
+  }
+  
+  @Post('/form')
+  @ApiOperation({ summary: 'Create a form' })
+  @ApiBearerAuth()
+  async createFormForModule (
+    @Body() formQuestions: formBatchDto
+  ){
+    const res = await this.contentService.createFormForModule(
+      formQuestions
+    );
+    return res;
+  }
+
+  @Get('/allFormQuestions')
+  @ApiOperation({ summary: 'Get all form Questions' })
+  @ApiQuery({
+    name: 'typeId',
+    required: false,
+    type: Number,
+    description: 'typeId',
+  })
+  @ApiQuery({
+    name: 'searchTerm',
+    required: false,
+    type: String,
+    description: 'Search by name or email',
+  })
+  @ApiBearerAuth()
+  async getAllFormQuestions(
+  @Query('typeId') typeId: number,
+  @Query('searchTerm') searchTerm: string,
+  ): Promise<object> {
+    const res = await this.contentService.getAllFormQuestions(
+    typeId,
+	  searchTerm,
+    );
+    return res;
+  }
+
```

Comment:
If we are creating the form then why the name is edit Form, if we are editing the form then why it is the POST method?

Reply:
I've taken reference from the earlier code and it has been like that ever since, usually editForm is the part where we mainly create the form that's why Post method

Reply:
Fine. Thank you